Add Certificate Authority Resource (Certificate Authority Service)

connorsmith-w opened the pull request:
<!-- Put a description of what this PR is for here, along with any references to issues that this resolves or contributes to -->

hashicorp/terraform-provider-google#7657

Adds "google_privateca_certificate_authority" (name is duplicated because "Certificate Authority" is also the product name)

See https://cloud.google.com/certificate-authority-service/docs/reference/rest/v1beta1/projects.locations.certificateAuthorities for resource documentation.

Notes:
- This change doesn't implement support for subordinate CAs, which require additional customization because they must be activated.

Customizations:
- Use POST :scheduleDelete to delete the resource (delete is not supported)
- On pre_delete, POST :disable to disable the resources (required for scheduling deletion)
- Check resource deletion by checking that status is DELETION_PENDING

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_privateca_certificate_authority`
```

modular-magician commented:
Hello!  I am a robot who works on Magic Modules PRs.

I have detected that you are a community contributor, so your PR will be assigned to someone with a commit-bit on this repo for initial review.

Thanks for your contribution!  A human will be with you soon.

@scottsuarez, please review this PR or find an appropriate assignee.

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4382-old..auto-pr-4382) ( 2 files changed, 361 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4382-old..auto-pr-4382) ( 8 files changed, 1857 insertions(+), 3 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-4382-old..auto-pr-4382) ( 8 files changed, 258 insertions(+))

drebes commented:
Hi, I have some ongoing work for Private CA [here](https://github.com/GoogleCloudPlatform/magic-modules/compare/master...drebes:privateca-resource), however I didn't have much time to progress recently. Would you like to pick up this work?

connorsmith-w commented:
@drebes Updated to use "privateca" as the base name as discussed out of band.

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4382-old..auto-pr-4382) ( 3 files changed, 710 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4382-old..auto-pr-4382) ( 11 files changed, 3698 insertions(+), 2 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-4382-old..auto-pr-4382) ( 16 files changed, 516 insertions(+))

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4382-old..auto-pr-4382) ( 3 files changed, 710 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4382-old..auto-pr-4382) ( 11 files changed, 3698 insertions(+), 2 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-4382-old..auto-pr-4382) ( 16 files changed, 516 insertions(+))

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4382-old..auto-pr-4382) ( 2 files changed, 361 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4382-old..auto-pr-4382) ( 8 files changed, 1857 insertions(+), 3 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-4382-old..auto-pr-4382) ( 8 files changed, 258 insertions(+))

ScottSuarez commented on products/privateca/api.yaml:
```diff
@@ -0,0 +1,265 @@
+        description: The Type of this CertificateAuthority.
+        input: true
+        values:
+          - :SELF_SIGNED
```
I'm assuming we don't have `SUBORDINATE` and `TYPE_UNSPECIFIED` as they are not supported in a terraform config at this time

connorsmith-w replied:
Right. I omitted SUBORDINATE for now because it has a more complex creation flow. So can be a followup PR if needed (if that's reasonable).

ScottSuarez replied:
sounds reasonable to me

ScottSuarez commented on products/privateca/api.yaml:
```diff
@@ -0,0 +1,265 @@
+        description: The Tier of this CertificateAuthority.
+        input: true
+        values:
+          - :ENTERPRISE
+          - :DEVOPS
```
`TIER_UNSPECIFIED`  ?

connorsmith-w replied:
TIER_UNSPECIFIED would cause the creation to fail (because it must be explicit); I thought maybe it made sense to enforce this choice with Terraform, but I'm not sure what the common practices is. Let me know if the full enum is preferable.

ScottSuarez replied:
this makes sense

ScottSuarez commented on products/privateca/api.yaml:
```diff
@@ -0,0 +1,265 @@
+              description: Contains only valid 32-bit IPv4 addresses or RFC 4291 IPv6 addresses.
+              item_type: Api::Type::String
+        - !ruby/object:Api::Type::NestedObject
+          name: 'reusableConfig'
```
isn't there a subproperty to this field for [reusableConfigValues](https://cloud.google.com/certificate-authority-service/docs/reference/rest/v1beta1/ReusableConfigWrapper). It says it's required on the documentation

connorsmith-w replied:
I believe this field is a oneof (per the doc: "config_values can be only one of the following": reusableconfig or reusableconfigvalues). I only added the simpler one here (as discussed out of band).

ScottSuarez replied:
ahh I see, thanks for the clarification :)

ScottSuarez commented on products/privateca/api.yaml:
```diff
@@ -0,0 +1,265 @@
+        input: true
+        properties:
+        - !ruby/object:Api::Type::NestedObject
+          name: 'subjectConfig'
```
shouldn't this have property [publicKey](https://cloud.google.com/certificate-authority-service/docs/reference/rest/v1beta1/CertificateConfig)

connorsmith-w replied:
Per the doc, I think that field is only used when creating Certificates as opposed to self-signed CAs (the config message is reused for both cases).

ScottSuarez commented on products/privateca/api.yaml:
```diff
@@ -0,0 +1,265 @@
+          Used when issuing certificates for this CertificateAuthority. If this CertificateAuthority
+          is a self-signed CertificateAuthority, this key is also used to sign the self-signed CA
+          certificate. Otherwise, it is used to sign a CSR.
+        required: true
```
```suggestion
        required: true
        input: true
```

ScottSuarez commented on products/privateca/api.yaml:
```diff
@@ -0,0 +1,265 @@
+        properties:
+        - !ruby/object:Api::Type::Enum
+          name: 'algorithm'
+          description: |
+            The algorithm to use for creating a managed Cloud KMS key for a for a simplified
+            experience. All managed keys will be have their ProtectionLevel as HSM.
+          values:
```
```suggestion
          description: |
            The algorithm to use for creating a managed Cloud KMS key for a for a simplified
            experience. All managed keys will be have their ProtectionLevel as HSM.
          required: true
          values:
```

ScottSuarez commented on products/privateca/api.yaml:
```diff
@@ -0,0 +1,265 @@
+          fractional digits, terminated by 's'. Example: "3.5s".
+        default_value: 315360000s  # 10 years
+      - !ruby/object:Api::Type::NestedObject
+        name: 'keySpec'
```
missing property [cloudKmsKeyVersion](https://cloud.google.com/certificate-authority-service/docs/reference/rest/v1beta1/projects.locations.certificateAuthorities#KeyVersionSpec) .... docs say this is required..

connorsmith-w replied:
I think this is another instance of the oneof marking both as required when only one should be specified (and I've ommitted the more advanced/complex config in this initial version).

ScottSuarez replied:
gotcha gotcha

ScottSuarez commented on products/privateca/api.yaml:
```diff
@@ -0,0 +1,265 @@
+          description: |
+            When true, includes a URL to the issuing CA certificate in the "authority
+            information access" X.509 extension.
+          default_value: true
```
```suggestion
          default_value: true
          required: true
```

ScottSuarez commented on products/privateca/api.yaml:
```diff
@@ -0,0 +1,265 @@
+            When true, includes a URL to the CRL corresponding to certificates issued from a
+            CertificateAuthority. CRLs will expire 7 days from their creation. However, we will
+            rebuild daily. CRLs are also rebuilt shortly after a certificate is revoked.
+          default_value: false
```
```suggestion
          default_value: false
          required: true
```

ScottSuarez commented on products/privateca/api.yaml:
```diff
@@ -0,0 +1,265 @@
+          output: true
+      - !ruby/object:Api::Type::KeyValuePairs
+        name: labels
+        description: Labels with user-defined metadata.
```
optional: add createTime and updateTime

connorsmith-w replied:
Added.

ScottSuarez reviewed (commented):
adding some comments... mainly just discrepancies between the live docs and your code. Some fields are stated as required in the docs and not included as parameters for the resource. I'm sure you know what you're doing though

ScottSuarez commented:
/gcbrun

modular-magician commented:
I have triggered VCR tests based on this PR's diffs. See the results here: "https://ci-oss.hashicorp.engineering/viewQueued.html?itemId=166875"

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4382-old..auto-pr-4382) ( 2 files changed, 361 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4382-old..auto-pr-4382) ( 7 files changed, 1859 insertions(+), 2 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-4382-old..auto-pr-4382) ( 8 files changed, 258 insertions(+))

modular-magician commented:
I have triggered VCR tests in RECORDING mode for the following tests that failed during VCR: TestAccInstanceTemplateDatasource_filter|TestAccInstanceTemplateDatasource_filter_mostRecent|TestAccActiveDirectoryDomainTrust_activeDirectoryDomainTrustBasicExample|TestAccApigeeOrganization_apigeeOrganizationCloudBasicTestExample|TestAccApigeeOrganization_apigeeOrganizationCloudFullTestExample|TestAccContainerCluster_withPrivateClusterConfigMissingCidrBlock|TestAccPrivatecaCertificateAuthority_privatecaCertificateAuthorityBasicExample|TestAccPrivatecaCertificateAuthority_privatecaCertificateAuthorityFullExample You can view the result here: "https://ci-oss.hashicorp.engineering/viewQueued.html?itemId=166877"

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4382-old..auto-pr-4382) ( 2 files changed, 361 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4382-old..auto-pr-4382) ( 7 files changed, 1860 insertions(+), 2 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-4382-old..auto-pr-4382) ( 8 files changed, 258 insertions(+))

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4382-old..auto-pr-4382) ( 2 files changed, 361 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4382-old..auto-pr-4382) ( 7 files changed, 1860 insertions(+), 2 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-4382-old..auto-pr-4382) ( 8 files changed, 258 insertions(+))

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4382-old..auto-pr-4382) ( 2 files changed, 361 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4382-old..auto-pr-4382) ( 7 files changed, 1860 insertions(+), 2 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-4382-old..auto-pr-4382) ( 8 files changed, 258 insertions(+))

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4382-old..auto-pr-4382) ( 2 files changed, 361 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4382-old..auto-pr-4382) ( 8 files changed, 1861 insertions(+), 3 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-4382-old..auto-pr-4382) ( 8 files changed, 258 insertions(+))

connorsmith-w commented:
@ScottSuarez marked more fields required (now that I understand what input and required mean :) ), added createtime/updateTime, and addressed other comments. Thank you for the thorough review!

ScottSuarez commented:
/gcbrun

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4382-old..auto-pr-4382) ( 2 files changed, 372 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4382-old..auto-pr-4382) ( 7 files changed, 1903 insertions(+), 2 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-4382-old..auto-pr-4382) ( 8 files changed, 258 insertions(+))

modular-magician commented:
I have triggered VCR tests based on this PR's diffs. See the results here: "https://ci-oss.hashicorp.engineering/viewQueued.html?itemId=166923"

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4382-old..auto-pr-4382) ( 2 files changed, 372 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4382-old..auto-pr-4382) ( 7 files changed, 1903 insertions(+), 2 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-4382-old..auto-pr-4382) ( 8 files changed, 258 insertions(+))

modular-magician commented:
I have triggered VCR tests in RECORDING mode for the following tests that failed during VCR: TestAccDataSourceComputeNetworkEndpointGroup|TestAccDataSourceComputeLbIpRanges_basic|TestAccDataSourceGoogleCloudRunLocations_basic|TestAccDataSourceCloudIdentityGroupMemberships_basic|TestAccDataSourceGoogleCloudRunService_basic|TestAccDataSourceGoogleCloudRunService_optionalProject|TestAccDataSourceCloudIdentityGroups_basic|TestAccDataSourceGameServicesGameServerDeploymentRollout_basic|TestAccDataSourceDnsManagedZone_basic|TestAccDataSourceDNSKeys_noDnsSec|TestAccDataSourceDNSKeys_basic|TestAccDataSourceGoogleActiveFolder_default|TestAccDataSourceGoogleActiveFolder_space|TestAccDataSourceGoogleAppEngineDefaultServiceAccount_basic|TestAccDataSourceGoogleBigqueryDefaultServiceAccount_basic|TestAccDataSourceGoogleBillingAccount_byFullName|TestAccDataSourceGoogleBillingAccount_byDisplayName|TestAccDataSourceGoogleBillingAccount_byShortName|TestAccDataSourceGoogleActiveFolder_dash|TestAccDataSourceGoogleBillingAccount_byFullNameClosed|TestAccDataSourceGoogleClientConfig_basic|TestAccDataSourceGoogleClientOpenIDUserinfo_basic|TestAccDataSourceComposerEnvironment_basic|TestAccDataSourceGoogleCloudFunctionsFunction_basic|TestAccDataSourceComputeBackendBucket_basic|TestAccDataSourceComputeAddress|TestAccDataSourceComputeBackendService_basic|TestAccDataSourceComposerImageVersions_basic|TestAccDataSourceGoogleComputeDefaultServiceAccount_basic|TestAccInstanceTemplateDatasource_filter_mostRecent|TestAccDataSourceGoogleForwardingRule|TestAccInstanceTemplateDatasource_filter|TestAccDataSourceComputeGlobalAddress|TestAccDataSourceComputeImageFilter|TestAccDataSourceComputeImage|TestAccDataSourceGoogleComputeInstanceGroup_basic|TestAccDataSourceGoogleComputeInstanceGroup_fromIGM|TestAccDataSourceGoogleComputeInstanceGroup_withNamedPort|TestAccInstanceTemplateDatasource_name|TestAccDataSourceComputeInstanceSerialPort_basic|TestAccDataSourceComputeSslCertificate|TestAccDataSourceComputeNodeTypes_basic|TestAccDataSourceGoogleNetwork|TestAccDataSourceComputeInstance_basic|TestAccDataSourceComputeRegionSslCertificate|TestAccComputeRegions_basic|TestAccDataSourceGoogleSslPolicy|TestAccDataSourceComputeResourcePolicy|TestAccDataSourceComputeRouter|TestAccDataSourceGoogleSubnetwork|TestAccComputeZones_filter|TestAccDataSourceGoogleVpnGateway|TestAccComputeZones_basic|TestAccContainerClusterDatasource_zonal|TestAccContainerEngineVersions_basic|TestAccContainerClusterDatasource_regional|TestAccDataSourceGoogleFolder_byShortName|TestAccDataSourceGoogleFolderOrganizationPolicy_basic|TestAccDataSourceGoogleFolder_byFullName|TestAccContainerEngineVersions_filtered|TestAccDataSourceGoogleGlobalForwardingRule|TestAccDataSourceIAMRole|TestAccDataSourceGoogleFolder_byFullNameNotFound|TestAccDataSourceGoogleIamTestablePermissions_basic|TestAccDataSourceGoogleFolder_lookupOrganization|TestAccDataSourceGoogleNetblockIpRanges_basic|TestAccDataSourceGoogleKmsCryptoKey_basic|TestAccDataSourceGoogleMonitoringUptimeCheckIps_basic|TestAccDataSourceGoogleOrganization_byShortName|TestAccDataSourceGoogleKmsKeyRing_basic|TestAccDataSourceGoogleOrganization_byFullName|TestAccDataKmsSecretCiphertext_basic|TestAccDataSourceGoogleOrganization_byDomain|TestAccDataSourceGoogleProjectOrganizationPolicy_basic|TestAccDataSourceGoogleKmsCryptoKeyVersion_basic|TestAccDataSourceGoogleProjects_basic|TestAccDataSourceGoogleProject_basic|TestAccRedisInstanceDatasource_basic|TestAccDataSourceGoogleServiceAccountIdToken_basic|TestAccDataSourceGoogleSQLCaCerts_basic|TestAccDatasourceGoogleServiceAccount_basic|TestAccDatasourceGoogleServiceAccountKey_basic|TestAccDataSourceGoogleServiceAccountAccessToken_basic|TestAccDataSourceGoogleStorageProjectServiceAccount_basic|TestAccDataSourceGoogleServiceAccountIdToken_impersonation|TestAccDataSourceGoogleStorageTransferProjectServiceAccount_basic|TestAccDataSourceIAMBetaWorkloadIdentityPoolProvider_basic|TestAccDataSourceIAMBetaWorkloadIdentityPool_basic|TestAccDataSourceGoogleMonitoringNotificationChannel_byDisplayName|TestAccDataSourceGoogleMonitoringNotificationChannel_byDisplayNameAndType|TestAccDataSourceGoogleMonitoringNotificationChannel_byTypeAndLabel|TestAccDataSourceGoogleMonitoringNotificationChannel_ErrorNoDisplayNameOrType|TestAccDataSourceGoogleMonitoringNotificationChannel_ErrorNotUnique|TestAccDataSourceGoogleMonitoringNotificationChannel_UserLabel|TestAccDataSourceGoogleMonitoringNotificationChannel_ErrorNotFound|TestAccDataSourceMonitoringService_AppEngine|TestAccDataSourceGooglePubsubTopic_basic|TestAccDatasourceSecretManagerSecretVersion_latest|TestAccDatasourceSecretManagerSecretVersion_basic|TestAccDataSourceGooglePubsubTopic_optionalProject You can view the result here: "https://ci-oss.hashicorp.engineering/viewQueued.html?itemId=166924"

connorsmith-w commented:
It seems terraform did not like having "required: true" set for fields which also have a default_value. Fixed (and verified that local test passes for me).

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4382-old..auto-pr-4382) ( 2 files changed, 375 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4382-old..auto-pr-4382) ( 7 files changed, 1909 insertions(+), 2 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-4382-old..auto-pr-4382) ( 8 files changed, 261 insertions(+))

ScottSuarez commented:
ahh cool cool, thanks for checking that ! sorry for the confusion I suppose we both learned something here ;)

/gcbrun

modular-magician commented:
I have triggered VCR tests based on this PR's diffs. See the results here: "https://ci-oss.hashicorp.engineering/viewQueued.html?itemId=167106"

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4382-old..auto-pr-4382) ( 2 files changed, 375 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4382-old..auto-pr-4382) ( 7 files changed, 1909 insertions(+), 2 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-4382-old..auto-pr-4382) ( 8 files changed, 261 insertions(+))

modular-magician commented:
I have triggered VCR tests in RECORDING mode for the following tests that failed during VCR: TestAccInstanceTemplateDatasource_filter|TestAccInstanceTemplateDatasource_filter_mostRecent|TestAccActiveDirectoryDomainTrust_activeDirectoryDomainTrustBasicExample|TestAccApigeeOrganization_apigeeOrganizationCloudBasicTestExample|TestAccApigeeOrganization_apigeeOrganizationCloudFullTestExample|TestAccCloudRunService_cloudRunServiceMultipleEnvironmentVariablesExample|TestAccPrivatecaCertificateAuthority_privatecaCertificateAuthorityBasicExample|TestAccPrivatecaCertificateAuthority_privatecaCertificateAuthorityFullExample You can view the result here: "https://ci-oss.hashicorp.engineering/viewQueued.html?itemId=167113"

connorsmith-w commented:
@ScottSuarez do you know why generate-diffs may be failing? I don't seem to have the read ACL for it.

ScottSuarez commented:
looks like some linter failures on the run

```
go: finding github.com/dnaeon/go-vcr v1.0.1
?   	github.com/hashicorp/terraform-provider-google-beta/v3	[no test files]
ok  	github.com/hashicorp/terraform-provider-google-beta/v3/google-beta	22.240s
?   	github.com/hashicorp/terraform-provider-google-beta/v3/scripts/affectedtests	[no test files]
?   	github.com/hashicorp/terraform-provider-google-beta/v3/scripts/sidebar	[no test files]
?   	github.com/hashicorp/terraform-provider-google-beta/v3/version	[no test files]
==> Checking source code against linters...
google-beta/resource_compute_instance_template_test.go:927:9: S1034: assigning the result of this type assertion to a variable (switch instanceTemplate := instanceTemplate.(type)) could eliminate type assertions in switch cases (gosimple)
	switch instanceTemplate.(type) {
	       ^
google-beta/resource_compute_instance_test.go:2086:9: S1034: assigning the result of this type assertion to a variable (switch instance := instance.(type)) could eliminate type assertions in switch cases (gosimple)
	switch instance.(type) {
	       ^
google-beta/common_operation.go:60:10: S1039: unnecessary use of fmt.Sprintf (gosimple)
		return fmt.Sprintf("Operation is nil!")
		       ^
google-beta/error_retry_predicates.go:295:17: S1039: unnecessary use of fmt.Sprintf (gosimple)
			return true, fmt.Sprintf("Retry on Function failing to pull code from GCS")
			             ^
google-beta/resource_secret_manager_secret_version.go:269:9: S1039: unnecessary use of fmt.Sprintf (gosimple)
		panic(fmt.Sprintf("Version name doesn not fit the format `projects/{{project}}/secrets/{{secret}}/versions{{version}}`"))
		      ^
google-beta/batcher_test.go:230:18: S1039: unnecessary use of fmt.Sprintf (gosimple)
			DebugId:      fmt.Sprintf("timeout test"),
			              ^
google-beta/data_source_monitoring_service_test.go:34:9: S1039: unnecessary use of fmt.Sprintf (gosimple)
	return fmt.Sprintf(`
	       ^
google-beta/resource_cloudbuild_trigger_test.go:214:9: S1039: unnecessary use of fmt.Sprintf (gosimple)
	return fmt.Sprintf(`
	       ^
google-beta/resource_compute_attached_disk_test.go:164:9: S1039: unnecessary use of fmt.Sprintf (gosimple)
	return fmt.Sprintf(`
	       ^
google-beta/resource_compute_attached_disk_test.go:173:9: S1039: unnecessary use of fmt.Sprintf (gosimple)
	return fmt.Sprintf(`
	       ^
google-beta/resource_compute_ssl_certificate_test.go:68:9: S1039: unnecessary use of fmt.Sprintf (gosimple)
	return fmt.Sprintf(`
	       ^
google-beta/resource_google_project_test.go:393:7: S1039: unnecessary use of fmt.Sprintf (gosimple)
	l += fmt.Sprintf("}\n}")
	     ^
google-beta/resource_monitoring_dashboard_test.go:146:9: S1039: unnecessary use of fmt.Sprintf (gosimple)
	return fmt.Sprintf(`
	       ^
google-beta/resource_monitoring_dashboard_test.go:166:9: S1039: unnecessary use of fmt.Sprintf (gosimple)
	return fmt.Sprintf(`
	       ^
google-beta/resource_monitoring_dashboard_test.go:234:9: S1039: unnecessary use of fmt.Sprintf (gosimple)
	return fmt.Sprintf(`
	       ^
google-beta/resource_monitoring_notification_channel_test.go:110:9: S1039: unnecessary use of fmt.Sprintf (gosimple)
	return fmt.Sprintf(`
	       ^
google-beta/resource_monitoring_notification_channel_test.go:141:9: S1039: unnecessary use of fmt.Sprintf (gosimple)
	return fmt.Sprintf(`
	       ^
google-beta/resource_monitoring_slo_test.go:487:9: S1039: unnecessary use of fmt.Sprintf (gosimple)
	return fmt.Sprintf(`
	       ^
google-beta/resource_monitoring_slo_test.go:499:9: S1039: unnecessary use of fmt.Sprintf (gosimple)
	return fmt.Sprintf(`
	       ^
google-beta/resource_monitoring_slo_test.go:523:9: S1039: unnecessary use of fmt.Sprintf (gosimple)
	return fmt.Sprintf(`
	       ^
google-beta/resource_monitoring_slo_test.go:546:9: S1039: unnecessary use of fmt.Sprintf (gosimple)
	return fmt.Sprintf(`
	       ^
google-beta/resource_monitoring_slo_test.go:570:9: S1039: unnecessary use of fmt.Sprintf (gosimple)
	return fmt.Sprintf(`
	       ^
google-beta/resource_monitoring_slo_test.go:595:9: S1039: unnecessary use of fmt.Sprintf (gosimple)
	return fmt.Sprintf(`
	       ^
google-beta/resource_monitoring_slo_test.go:632:9: S1039: unnecessary use of fmt.Sprintf (gosimple)
	return fmt.Sprintf(`
	       ^
google-beta/node_config.go:624:2: S1008: should use 'return len(labels) <= 0' instead of 'if len(labels) > 0 { return false }; return true' (gosimple)
	if len(labels) > 0 {
	^
google-beta/node_config.go:692:2: S1008: should use 'return len(taintSet) <= 0' instead of 'if len(taintSet) > 0 { return false }; return true' (gosimple)
	if len(taintSet) > 0 {
	^
google-beta/resource_secret_manager_secret_version.go:279:2: S1008: should use 'return v.(string) == "ENABLED"' instead of 'if v.(string) == "ENABLED" { return true }; return false' (gosimple)
	if v.(string) == "ENABLED" {
	^
google-beta/resource_cloud_run_service.go:39:41: S1007: should use raw string (`...`) with regexp.MustCompile to avoid having to escape twice (gosimple)
var cloudRunGoogleProvidedAnnotations = regexp.MustCompile("serving\\.knative\\.dev/(?:(?:creator)|(?:lastModifier))$")
                                        ^
google-beta/data_source_google_monitoring_uptime_check_ips_test.go:19:40: S1007: should use raw string (`...`) with regexp.MustCompile to avoid having to escape twice (gosimple)
						"uptime_check_ips.0.ip_address", regexp.MustCompile("^\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}$")),
						                                 ^
google-beta/error_retry_predicates.go:170:3: S1021: should merge variable declaration with assignment on next line (gosimple)
		var ierr interface{}
		^
google-beta/iam_big_query_table.go:142:2: S1021: should merge variable declaration with assignment on next line (gosimple)
	var obj map[string]interface{}
	^
google-beta/iam_iap_app_engine_service.go:142:2: S1021: should merge variable declaration with assignment on next line (gosimple)
	var obj map[string]interface{}
	^
google-beta/iam_iap_app_engine_version.go:157:2: S1021: should merge variable declaration with assignment on next line (gosimple)
	var obj map[string]interface{}
	^
google-beta/iam_iap_tunnel.go:113:2: S1021: should merge variable declaration with assignment on next line (gosimple)
	var obj map[string]interface{}
	^
google-beta/iam_iap_tunnel_instance.go:151:2: S1021: should merge variable declaration with assignment on next line (gosimple)
	var obj map[string]interface{}
	^
google-beta/iam_iap_web.go:113:2: S1021: should merge variable declaration with assignment on next line (gosimple)
	var obj map[string]interface{}
	^
google-beta/iam_iap_web_backend_service.go:127:2: S1021: should merge variable declaration with assignment on next line (gosimple)
	var obj map[string]interface{}
	^
google-beta/iam_iap_web_type_app_engine.go:141:2: S1021: should merge variable declaration with assignment on next line (gosimple)
	var obj map[string]interface{}
	^
google-beta/iam_iap_web_type_compute.go:113:2: S1021: should merge variable declaration with assignment on next line (gosimple)
	var obj map[string]interface{}
	^
google-beta/resource_bigquery_data_transfer_config.go:843:2: S1021: should merge variable declaration with assignment on next line (gosimple)
	var params map[string]string
	^
google-beta/resource_binary_authorization_policy.go:440:2: S1021: should merge variable declaration with assignment on next line (gosimple)
	var obj map[string]interface{}
	^
google-beta/resource_compute_disk_resource_policy_attachment.go:226:2: S1021: should merge variable declaration with assignment on next line (gosimple)
	var obj map[string]interface{}
	^
google-beta/resource_compute_region_disk_resource_policy_attachment.go:226:2: S1021: should merge variable declaration with assignment on next line (gosimple)
	var obj map[string]interface{}
	^
google-beta/resource_google_project.go:115:2: S1021: should merge variable declaration with assignment on next line (gosimple)
	var pid string
	^
google-beta/resource_monitoring_notification_channel.go:567:2: S1021: should merge variable declaration with assignment on next line (gosimple)
	var labels map[string]string
	^
google-beta/iam_bigquery_dataset.go:249:9: S1025: the argument is already a string, there's no need to use fmt.Sprintf (gosimple)
	return fmt.Sprintf("%s", u.datasetId)
	       ^
google-beta/iam_data_catalog_policy_tag.go:148:61: S1025: the argument is already a string, there's no need to use fmt.Sprintf (gosimple)
	urlTemplate := fmt.Sprintf("{{DataCatalogBasePath}}%s:%s", fmt.Sprintf("%s", u.policyTag), methodIdentifier)
	                                                           ^
google-beta/iam_data_catalog_policy_tag.go:157:9: S1025: the argument is already a string, there's no need to use fmt.Sprintf (gosimple)
	return fmt.Sprintf("%s", u.policyTag)
	       ^
google-beta/iam_service_directory_namespace.go:148:66: S1025: the argument is already a string, there's no need to use fmt.Sprintf (gosimple)
	urlTemplate := fmt.Sprintf("{{ServiceDirectoryBasePath}}%s:%s", fmt.Sprintf("%s", u.name), methodIdentifier)
	                                                                ^
google-beta/iam_service_directory_namespace.go:157:9: S1025: the argument is already a string, there's no need to use fmt.Sprintf (gosimple)
	return fmt.Sprintf("%s", u.name)
	       ^
google-beta/common_diff_suppress.go:86:9: SA6005: should use strings.EqualFold instead (staticcheck)
	return strings.ToUpper(old) == strings.ToUpper(new)
	       ^
google-beta/resource_compute_firewall.go:52:9: SA6005: should use strings.EqualFold instead (staticcheck)
	return strings.ToLower(old) == strings.ToLower(new)
	       ^
google-beta/resource_dns_record_set.go:53:13: SA6005: should use strings.EqualFold instead (staticcheck)
					return strings.ToLower(strings.Trim(old, `"`)) == strings.ToLower(strings.Trim(new, `"`))
					       ^
google-beta/resource_iam_member.go:112:9: SA6005: should use strings.EqualFold instead (staticcheck)
					if strings.ToLower(m) == strings.ToLower(member) {
					   ^
google-beta/resource_iam_member.go:237:7: SA6005: should use strings.EqualFold instead (staticcheck)
			if strings.ToLower(m) == strings.ToLower(eMember.Members[0]) {
			   ^
google-beta/validation.go:280:34: SA6005: should use strings.EqualFold instead (staticcheck)
			if v == str || (ignoreCase && strings.ToLower(v) == strings.ToLower(str)) {
			                              ^
google-beta/common_operation.go:154:16: SA1019: c.WaitForState is deprecated: Please use WaitForStateContext to ensure proper plugin shutdown  (staticcheck)
	opRaw, err := c.WaitForState()
	              ^
google-beta/common_polling.go:40:10: SA1019: resource.Retry is deprecated: Please use RetryContext to ensure proper plugin shutdown  (staticcheck)
		return resource.Retry(timeout, func() *resource.RetryError {
		       ^
google-beta/common_polling.go:87:16: SA1019: c.WaitForState is deprecated: Please use WaitForStateContext to ensure proper plugin shutdown  (staticcheck)
	_, waitErr := c.WaitForState()
	              ^
google-beta/data_source_google_billing_account.go:54:18: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	open, openOk := d.GetOkExists("open")
	                ^
google-beta/provider.go:1400:17: SA1019: schema.StopContext is deprecated: The use of a global context is discouraged. Please use the new context aware CRUD methods.  (staticcheck)
	stopCtx, ok := schema.StopContext(ctx)
	               ^
google-beta/resource_access_approval_folder_settings.go:169:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("notification_emails"); !isEmptyValue(reflect.ValueOf(notificationEmailsProp)) && (ok || !reflect.DeepEqual(v, notificationEmailsProp)) {
	                   ^
google-beta/resource_access_approval_folder_settings.go:175:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("enrolled_services"); !isEmptyValue(reflect.ValueOf(enrolledServicesProp)) && (ok || !reflect.DeepEqual(v, enrolledServicesProp)) {
	                   ^
google-beta/resource_access_approval_folder_settings.go:280:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("notification_emails"); !isEmptyValue(reflect.ValueOf(v)) && (ok || !reflect.DeepEqual(v, notificationEmailsProp)) {
	                   ^
google-beta/resource_access_approval_folder_settings.go:286:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("enrolled_services"); !isEmptyValue(reflect.ValueOf(v)) && (ok || !reflect.DeepEqual(v, enrolledServicesProp)) {
	                   ^
google-beta/resource_access_approval_organization_settings.go:131:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("notification_emails"); !isEmptyValue(reflect.ValueOf(notificationEmailsProp)) && (ok || !reflect.DeepEqual(v, notificationEmailsProp)) {
	                   ^
google-beta/resource_access_approval_organization_settings.go:137:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("enrolled_services"); !isEmptyValue(reflect.ValueOf(enrolledServicesProp)) && (ok || !reflect.DeepEqual(v, enrolledServicesProp)) {
	                   ^
google-beta/resource_access_approval_organization_settings.go:242:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("notification_emails"); !isEmptyValue(reflect.ValueOf(v)) && (ok || !reflect.DeepEqual(v, notificationEmailsProp)) {
	                   ^
google-beta/resource_access_approval_organization_settings.go:248:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("enrolled_services"); !isEmptyValue(reflect.ValueOf(v)) && (ok || !reflect.DeepEqual(v, enrolledServicesProp)) {
	                   ^
google-beta/resource_access_approval_project_settings.go:137:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("notification_emails"); !isEmptyValue(reflect.ValueOf(notificationEmailsProp)) && (ok || !reflect.DeepEqual(v, notificationEmailsProp)) {
	                   ^
google-beta/resource_access_approval_project_settings.go:143:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("enrolled_services"); !isEmptyValue(reflect.ValueOf(enrolledServicesProp)) && (ok || !reflect.DeepEqual(v, enrolledServicesProp)) {
	                   ^
google-beta/resource_access_approval_project_settings.go:149:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("project"); !isEmptyValue(reflect.ValueOf(projectProp)) && (ok || !reflect.DeepEqual(v, projectProp)) {
	                   ^
google-beta/resource_access_approval_project_settings.go:261:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("notification_emails"); !isEmptyValue(reflect.ValueOf(v)) && (ok || !reflect.DeepEqual(v, notificationEmailsProp)) {
	                   ^
google-beta/resource_access_approval_project_settings.go:267:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("enrolled_services"); !isEmptyValue(reflect.ValueOf(v)) && (ok || !reflect.DeepEqual(v, enrolledServicesProp)) {
	                   ^
google-beta/resource_access_approval_project_settings.go:273:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("project"); !isEmptyValue(reflect.ValueOf(v)) && (ok || !reflect.DeepEqual(v, projectProp)) {
	                   ^
google-beta/resource_access_context_manager_access_level.go:295:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("title"); !isEmptyValue(reflect.ValueOf(titleProp)) && (ok || !reflect.DeepEqual(v, titleProp)) {
	                   ^
google-beta/resource_access_context_manager_access_level.go:301:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("description"); !isEmptyValue(reflect.ValueOf(descriptionProp)) && (ok || !reflect.DeepEqual(v, descriptionProp)) {
	                   ^
google-beta/resource_access_context_manager_access_level.go:307:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("basic"); !isEmptyValue(reflect.ValueOf(basicProp)) && (ok || !reflect.DeepEqual(v, basicProp)) {
	                   ^
google-beta/resource_access_context_manager_access_level.go:313:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("custom"); !isEmptyValue(reflect.ValueOf(customProp)) && (ok || !reflect.DeepEqual(v, customProp)) {
	                   ^
google-beta/resource_access_context_manager_access_level.go:319:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("parent"); !isEmptyValue(reflect.ValueOf(parentProp)) && (ok || !reflect.DeepEqual(v, parentProp)) {
	                   ^
google-beta/resource_access_context_manager_access_level.go:325:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("name"); !isEmptyValue(reflect.ValueOf(nameProp)) && (ok || !reflect.DeepEqual(v, nameProp)) {
	                   ^
google-beta/resource_access_context_manager_access_level.go:443:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("title"); !isEmptyValue(reflect.ValueOf(v)) && (ok || !reflect.DeepEqual(v, titleProp)) {
	                   ^
google-beta/resource_access_context_manager_access_level.go:449:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("description"); !isEmptyValue(reflect.ValueOf(v)) && (ok || !reflect.DeepEqual(v, descriptionProp)) {
	                   ^
google-beta/resource_access_context_manager_access_level.go:455:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("basic"); !isEmptyValue(reflect.ValueOf(v)) && (ok || !reflect.DeepEqual(v, basicProp)) {
	                   ^
google-beta/resource_access_context_manager_access_level.go:461:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("custom"); !isEmptyValue(reflect.ValueOf(v)) && (ok || !reflect.DeepEqual(v, customProp)) {
	                   ^
google-beta/resource_access_context_manager_access_level_condition.go:209:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("ip_subnetworks"); !isEmptyValue(reflect.ValueOf(ipSubnetworksProp)) && (ok || !reflect.DeepEqual(v, ipSubnetworksProp)) {
	                   ^
google-beta/resource_access_context_manager_access_level_condition.go:215:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("required_access_levels"); !isEmptyValue(reflect.ValueOf(requiredAccessLevelsProp)) && (ok || !reflect.DeepEqual(v, requiredAccessLevelsProp)) {
	                   ^
google-beta/resource_access_context_manager_access_level_condition.go:221:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("members"); !isEmptyValue(reflect.ValueOf(membersProp)) && (ok || !reflect.DeepEqual(v, membersProp)) {
	                   ^
google-beta/resource_access_context_manager_access_level_condition.go:227:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("negate"); !isEmptyValue(reflect.ValueOf(negateProp)) && (ok || !reflect.DeepEqual(v, negateProp)) {
	                   ^
google-beta/resource_access_context_manager_access_level_condition.go:233:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("device_policy"); !isEmptyValue(reflect.ValueOf(devicePolicyProp)) && (ok || !reflect.DeepEqual(v, devicePolicyProp)) {
	                   ^
google-beta/resource_access_context_manager_access_level_condition.go:239:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("regions"); !isEmptyValue(reflect.ValueOf(regionsProp)) && (ok || !reflect.DeepEqual(v, regionsProp)) {
	                   ^
google-beta/resource_access_context_manager_access_levels.go:306:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("access_levels"); !isEmptyValue(reflect.ValueOf(accessLevelsProp)) && (ok || !reflect.DeepEqual(v, accessLevelsProp)) {
	                   ^
google-beta/resource_access_context_manager_access_levels.go:394:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("access_levels"); !isEmptyValue(reflect.ValueOf(v)) && (ok || !reflect.DeepEqual(v, accessLevelsProp)) {
	                   ^
google-beta/resource_access_context_manager_access_policy.go:88:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("parent"); !isEmptyValue(reflect.ValueOf(parentProp)) && (ok || !reflect.DeepEqual(v, parentProp)) {
	                   ^
google-beta/resource_access_context_manager_access_policy.go:94:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("title"); !isEmptyValue(reflect.ValueOf(titleProp)) && (ok || !reflect.DeepEqual(v, titleProp)) {
	                   ^
google-beta/resource_access_context_manager_access_policy.go:219:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("title"); !isEmptyValue(reflect.ValueOf(v)) && (ok || !reflect.DeepEqual(v, titleProp)) {
	                   ^
google-beta/resource_access_context_manager_gcp_user_access_binding.go:88:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("group_key"); !isEmptyValue(reflect.ValueOf(groupKeyProp)) && (ok || !reflect.DeepEqual(v, groupKeyProp)) {
	                   ^
google-beta/resource_access_context_manager_gcp_user_access_binding.go:94:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("access_levels"); !isEmptyValue(reflect.ValueOf(accessLevelsProp)) && (ok || !reflect.DeepEqual(v, accessLevelsProp)) {
	                   ^
google-beta/resource_access_context_manager_gcp_user_access_binding.go:201:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("access_levels"); !isEmptyValue(reflect.ValueOf(v)) && (ok || !reflect.DeepEqual(v, accessLevelsProp)) {
	                   ^
google-beta/resource_access_context_manager_service_perimeter.go:297:21: SA1019: d.GetOkExists is deprecated: usage is discouraged due to undefined behaviors and may be removed in a future version of the SDK  (staticcheck)
	} else if v, ok := d.GetOkExists("title"); !isEmptyValue(reflect.ValueOf(titleProp)) && (ok || !reflect.DeepEqual(v, titleProp)) {
	                   ^
google-beta/resource_container_node_pool_test.go:1701:6: func `testAccContainerNodePool_updateVersion` is unused (unused)
google-beta/resource_container_cluster_test.go:3336:6: func `testAccContainerCluster_withNodePoolUpdateVersion` is unused (unused)
google-beta/resource_container_cluster_test.go:2911:6: func `testAccContainerCluster_updateVersion` is unused (unused)
google-beta/resource_container_cluster_test.go:3315:6: func `testAccContainerCluster_withNodePoolLowerVersion` is unused (unused)
google-beta/resource_container_cluster_test.go:2896:6: func `testAccContainerCluster_withLowerVersion` is unused (unused)
google-beta/resource_container_node_pool_test.go:1677:6: func `testAccContainerNodePool_version` is unused (unused)
make: *** [lint] Error 1
GNUmakefile:27: recipe for target 'lint' failed
```

ScottSuarez commented:
> @ScottSuarez do you know why generate-diffs may be failing? I don't seem to have the read ACL for it.

i'm looking now... looks like linter rules. I'm asking my colleges

ScottSuarez commented:
linting was broken for a 7m period due to a bad merge

/gcbrun

modular-magician commented:
I have triggered VCR tests based on this PR's diffs. See the results here: "https://ci-oss.hashicorp.engineering/viewQueued.html?itemId=167242"

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4382-old..auto-pr-4382) ( 2 files changed, 375 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4382-old..auto-pr-4382) ( 7 files changed, 1909 insertions(+), 2 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-4382-old..auto-pr-4382) ( 8 files changed, 261 insertions(+))

ScottSuarez commented:
/gcbrun

modular-magician commented:
I have triggered VCR tests based on this PR's diffs. See the results here: "https://ci-oss.hashicorp.engineering/viewQueued.html?itemId=167303"

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4382-old..auto-pr-4382) ( 2 files changed, 375 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4382-old..auto-pr-4382) ( 7 files changed, 1909 insertions(+), 2 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-4382-old..auto-pr-4382) ( 8 files changed, 261 insertions(+))

modular-magician commented:
I have triggered VCR tests in RECORDING mode for the following tests that failed during VCR: TestAccCloudRunServiceIamBindingGenerated|TestAccCloudRunServiceIamMemberGenerated|TestAccCloudRunServiceIamPolicyGenerated|TestAccActiveDirectoryDomainTrust_activeDirectoryDomainTrustBasicExample|TestAccApigeeOrganization_apigeeOrganizationCloudBasicTestExample|TestAccApigeeOrganization_apigeeOrganizationCloudFullTestExample|TestAccCloudRunDomainMapping_cloudRunDomainMappingBasicExample|TestAccCloudRunService_cloudRunServiceBasicExample|TestAccCloudRunService_cloudRunServiceMultipleEnvironmentVariablesExample|TestAccCloudRunService_cloudRunServiceSqlExample|TestAccCloudRunService_cloudRunServiceNoauthExample|TestAccComputeHealthCheck_tcp_update|TestAccComputeHealthCheck_typeTransition|TestAccComputeRegionBackendService_basic|TestAccComputeRegionNetworkEndpointGroup_regionNetworkEndpointGroupCloudrunExample|TestAccPrivatecaCertificateAuthority_privatecaCertificateAuthorityBasicExample|TestAccPrivatecaCertificateAuthority_privatecaCertificateAuthorityFullExample You can view the result here: "https://ci-oss.hashicorp.engineering/viewQueued.html?itemId=167306"

ScottSuarez commented:
/gcbrun

had to enable the api on our test product. rerunning

modular-magician commented:
I have triggered VCR tests based on this PR's diffs. See the results here: "https://ci-oss.hashicorp.engineering/viewQueued.html?itemId=167311"

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4382-old..auto-pr-4382) ( 2 files changed, 375 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4382-old..auto-pr-4382) ( 7 files changed, 1909 insertions(+), 2 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-4382-old..auto-pr-4382) ( 8 files changed, 261 insertions(+))

modular-magician commented:
I have triggered VCR tests in RECORDING mode for the following tests that failed during VCR: TestAccActiveDirectoryDomainTrust_activeDirectoryDomainTrustBasicExample|TestAccApigeeOrganization_apigeeOrganizationCloudBasicTestExample|TestAccApigeeOrganization_apigeeOrganizationCloudFullTestExample|TestAccCloudRunService_cloudRunServiceMultipleEnvironmentVariablesExample|TestAccPrivatecaCertificateAuthority_privatecaCertificateAuthorityBasicExample|TestAccPrivatecaCertificateAuthority_privatecaCertificateAuthorityFullExample You can view the result here: "https://ci-oss.hashicorp.engineering/viewQueued.html?itemId=167319"

drebes commented:
It would be great if we could hold release for this until GoogleCloudPlatform/magic-modules#4407 is also merged.
